controlplane/rosa/api/v1beta2/rosacontrolplane_types.go

```diff
@@ -76,6 +77,169 @@ type RosaControlPlaneSpec struct { //nolint: maligned
 	// - ocmApiUrl: Optional, defaults to 'https://api.openshift.com'
 	// +optional
 	CredentialsSecretRef *corev1.LocalObjectReference `json:"credentialsSecretRef,omitempty"`
+
+	// PodCIDR
+	// +kubebuilder:validation:Format=cidr
+	// +optional
+	PodCIDR string `json:"PodCIDR,omitempty"`
+
+	// ServiceCIDR
+	// +kubebuilder:validation:Format=cidr
+	// +optional
+	ServiceCIDR string `json:"serviceCIDR,omitempty"`
+
+	// HostPrefix
+	// +optional
+	HostPrefix int `json:"hostPrefix,omitempty"`
```

Comment:
let's group `MachineCIRD`, `PodCIDR`, `ServiceCIDR` and `HostPrefix`  in a new struct `NetworkSpec` and add it as an optional field `network`